wellcomecollection/wellcomecollection.org
# View transitions

## What does this change?
Adds a `TransitionLink` component that is a wrapper around the `next/link` component that allows for the platform–native [View Transitions](https://developer.mozilla.org/en-US/docs/Web/API/View_Transitions_API) to run if they're available.

https://github.com/user-attachments/assets/7d9cb7c5-f618-40fa-a762-0c3bd13f9a9e

## How to test
Set `toggle_viewTransitions` to true locally and visit e.g. a [stops audio index](http://localhost:3000/guides/exhibitions/ZsdFlRIAACIAqKGB/audio-without-descriptions) – check zoom transitions work from cards to players, and slide transitions work with the previous/next buttons from individual stops.

## How can we measure success?
Possibly requires user testing

## Have we considered potential risks?
If the animation is inconsistent or janky, the experience could feel broken. ~This updates all `CardOuter` components which will then by default crossfade to their linked page. Not sure if this is desirable so maybe we need to make it conditional (or another component)~ <– made a separate component
